crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -33,6 +33,9 @@ class ApplicationController < ActionController::Base
     Crowbar::Installer.successful? || \
     Rails.env.test?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/lib/crowbar/upgrade/progress.yml")
```

Comment:
I think adding separate `running` file is the best of options so far. I don't like removing the status file after upgrade is `finished` as this would prevent us from even noticing this. As soon as the `current_step` would be set to `finished` the status API would stop responding.
Removing the `running` file should be pretty simple as it can be bound to step transition in status library. Removing the whole state of status file library is IMO more fragile.